modules/ding_nodelist/ding_nodelist.module

```diff
@@ -0,0 +1,2477 @@
+define('NL_WIDGET_TAXONOMY_LIKE', 'taxonomy');
+
+define('NL_TPL_ACTIVE', 0);
+define('NL_TPL_HIDDEN', 1);
+
+define('NL_CACHE_LIFETIME', 1800);
+define('NL_TITLE_SIZE', 25);
+define('NL_BODY_SIZE', 100);
+
+/**
+ * Implements hook_menu().
+ */
+function ding_nodelist_menu() {
+  $items = array();
+
+  $items['admin/config/ding/ding_nodelist'] = array(
+    'title' => 'Ding nodelist',
+    'description' => 'View and customize lists of content.',
+    'page callback' => 'drupal_get_form',
+    'page arguments' => array('ding_nodelist_settings_form'),
+    'access arguments' => array('configure nodelist'),
+    'type' => MENU_NORMAL_ITEM,
+    'file' => 'ding_nodelist.admin.inc',
+  );
+
+  $items['admin/config/ding/ding_nodelist/settings'] = array(
+    'title' => 'Settings',
+    'description' => 'Configure general settings.',
+    'tab_parent' => 'admin/config/ding/ding_nodelist',
+    'type' => MENU_DEFAULT_LOCAL_TASK,
+    'file' => 'ding_nodelist.admin.inc',
+  );
+
+  $items['admin/config/ding/ding_nodelist/templates'] = array(
+    'title' => 'Templates',
+    'description' => 'Manage content templates.',
+    'page callback' => 'drupal_get_form',
+    'page arguments' => array('ding_nodelist_templates_form'),
+    'access arguments' => array('configure nodelist'),
+    'type' => MENU_LOCAL_TASK,
+    'file' => 'ding_nodelist.admin.inc',
+  );
+
+  $items['admin/config/ding/ding_nodelist/trim_content'] = array(
+    'title' => 'Trim content',
+    'description' => 'Define title and teaser trim parameters for each widget',
+    'page callback' => 'drupal_get_form',
+    'page arguments' => array('ding_nodelist_trim_content'),
+    'access arguments' => array('configure nodelist'),
+    'type' => MENU_LOCAL_TASK,
+    'file' => 'ding_nodelist.admin.inc',
+  );
+
+  $items['ding_nodelist/autocomplete/%'] = array(
+    'page callback' => 'ding_nodelist_autocomplete',
+    'page arguments' => array(2),
+    'access arguments' => array('access content'),
+    'type' => MENU_CALLBACK,
+  );
+
+  return $items;
+}
+
+/**
+ * Implements hook_permission().
+ */
+function ding_nodelist_permission() {
+  return array(
+    'configure nodelist' => array(
+      'title' => t('Configure nodelist'),
+      'description' => t('Configure the behavior of nodelist'),
+    ),
+    'nodelist cache settings' => array(
+      'title' => t('Nodelist cache'),
+      'description' => t('Manage nodelist cache settings'),
+    ),
+  );
+}
+
+/**
+ * Implements hook_ctools_plugin_directory().
+ */
+function ding_nodelist_ctools_plugin_directory($module, $plugin) {
+  // we'll be nice and limit scandir() calls.
+  if ($module == 'ctools' && $plugin == 'content_types') {
+    return 'plugins/content_types';
+  }
+}
+
+/**
+ * Implements hook_ajax_render_alter().
+ *
+ * This method is a workaround to impossiblity
+ * of telling Drupal the order of CSS files in order to
+ * cascade styles properly. See the link.
+ *
+ * @link http://drupal.org/node/561858#comment-3171846
```

Comment:
I appreciate the reference to the issue but is there a reason why the link points to a specific comment? For me the comment provides little information about the problem at hand and as far as I can tell the attached patch is not used either.

![tests added fix drupal_add_js and drupal_add_css to work for ajax requests too by adding lazy-load to ajax framework 56 2016-09-14 11-09-59](https://cloud.githubusercontent.com/assets/73966/18506320/c8ac7aa6-7a6b-11e6-9336-96bef6505cea.png)